esphome/components/max6921/display.cpp

```diff
@@ -0,0 +1,953 @@
+    SEG_C | SEG_D | SEG_E | SEG_F | SEG_G,                  // 'b', (0x62)
+    SEG_D | SEG_E | SEG_G,                                  // 'c', (0x63)
+    SEG_B | SEG_C | SEG_D | SEG_E | SEG_G,                  // 'd', (0x64)
+    SEG_A | SEG_D | SEG_E | SEG_F | SEG_G,                  // 'e', (0x65)
+    SEG_A | SEG_E | SEG_F | SEG_G,                          // 'f', (0x66)
+    SEG_A | SEG_C | SEG_D | SEG_E | SEG_F,                  // 'g', (0x67)
+    SEG_C | SEG_E | SEG_F | SEG_G,                          // 'h', (0x68)
+    SEG_C,                                                  // 'i', (0x69)
+    SEG_B | SEG_C | SEG_D | SEG_E,                          // 'j', (0x6A)
+    SEG_UNSUPPORTED_CHAR,                                   // 'k', (0x6B)
+    SEG_D | SEG_E | SEG_F,                                  // 'l', (0x6C)
+    SEG_UNSUPPORTED_CHAR,                                   // 'm', (0x6D)
+    SEG_C | SEG_E | SEG_G,                                  // 'n', (0x6E)
+    SEG_C | SEG_D | SEG_E | SEG_G,                          // 'o', (0x6F)
+    SEG_A | SEG_B | SEG_E | SEG_F | SEG_G,                  // 'p', (0x70)
+    SEG_UNSUPPORTED_CHAR,                                   // 'q', (0x71)
+    SEG_E | SEG_G,                                          // 'r', (0x72)
+    SEG_A | SEG_C | SEG_D | SEG_F | SEG_G,                  // 's', (0x73)
+    SEG_UNSUPPORTED_CHAR,                                   // 't', (0x74)
+    SEG_C | SEG_D | SEG_E,                                  // 'u', (0x75)
+    SEG_UNSUPPORTED_CHAR,                                   // 'v', (0x76)
+    SEG_UNSUPPORTED_CHAR,                                   // 'w', (0x77)
+    SEG_UNSUPPORTED_CHAR,                                   // 'x', (0x78)
+    SEG_B | SEG_E | SEG_F | SEG_G,                          // 'y', (0x79)
+    SEG_UNSUPPORTED_CHAR,                                   // 'z', (0x7A)
+    SEG_B | SEG_C | SEG_G,                                  // '{', (0x7B)
+    SEG_UNSUPPORTED_CHAR,                                   // '|', (0x7C)
+    SEG_E | SEG_F | SEG_G,                                  // '}', (0x7D)
+    SEG_UNSUPPORTED_CHAR,                                   // '~', (0x7E)
+};
+
+void Max6921Display::setup(std::vector<uint8_t> &seg_to_out_map, std::vector<uint8_t> &pos_to_out_map) {
+  this->default_update_interval_ = this->max6921_->get_update_interval();
+  this->seg_to_out_map_ = seg_to_out_map;
+  this->pos_to_out_map_ = pos_to_out_map;
+  this->num_digits_ = pos_to_out_map.size();
+  ESP_LOGCONFIG(TAG, "Display digits: %u", this->num_digits_);
+
+  // setup font...
+  this->init_font_();
+
+  // display output buffer...
+  this->out_buf_size_ = this->num_digits_ * 3;
+  this->out_buf_ = new uint8_t[this->out_buf_size_];  // NOLINT
+  memset(this->out_buf_, 0, this->out_buf_size_);
+
+  // find smallest segment DOUT number...
+  this->seg_out_smallest_ = 19;
+  for (unsigned char i : this->seg_to_out_map_) {
```

Comment:
We do not use `unsigned char` 
```suggestion
  for (uint8_t i : this->seg_to_out_map_) {
```